template.go

```diff
@@ -54,15 +64,51 @@ func readYamlRequest(opts *Options) error {
 		return err
 	}
 
-	headers, err := yaml.Marshal(t.Headers)
-	if err != nil {
-		return err
+	if t.Peer != "" {
+		opts.TOpts.HostPorts = []string{t.Peer}
+	} else if t.Peers != nil {
+		opts.TOpts.HostPorts = t.Peers
+	} else if t.PeerList != "" {
+		opts.TOpts.HostPortFile = path.Join(path.Dir(opts.ROpts.YamlTemplate), t.PeerList)
+	}
+
+	// Header arguments have precedence over template headers.
+	if t.Headers != nil {
+		if opts.ROpts.Headers == nil {
+			opts.ROpts.Headers = t.Headers
+		} else {
+			for k, v := range t.Headers {
+				if _, exists := opts.ROpts.Headers[k]; !exists {
+					opts.ROpts.Headers[k] = v
+				}
+			}
+		}
+	}
```

Comment:
let's move this into a separate function, and unit test it individually. I think we can simplify this method, but want to get unit test coverage to make sure we don't break anything:
```go
if opts.ROpts.Headers == nil {
  opts.ROpts.Headers = t.Headers
} else {
  for k, v := range t.Headers {
    if _, exists := opts.ROpts.Headers[k]; !exists {
      opts.ROpts.Headers[k] = v
    }   
  }
}